node/src/components/consensus/protocols/highway.rs

```diff
@@ -585,15 +585,46 @@ where
                 }
             }
             Ok(HighwayMessage::RequestLatestState(panorama)) => {
+                trace!("received a request for the latest state");
                 let state = self.highway.state();
 
-                let merged_panorama = state.panorama().merge(state, &panorama);
+                let create_message =
+                    |observations: (&Observation<C>, &Observation<C>)| match observations {
+                        (Observation::None, Observation::None)
+                        | (Observation::Faulty, _)
+                        | (_, Observation::Faulty) => None,
```

Comment:
I'd first get the
```rust
(obs0, obs1) if obs0 == obs1 => None,
```
case out of the way. In the
```rust
(Observation::Faulty, _)
```
case we could send them our evidence, and in
```rust
(_, Observation::Faulty)
```
request it from them (`Dependency::Evidence`).